lib/api/core/models/security/profile.js

```diff
@@ -21,23 +38,33 @@ function Profile () {
 Profile.prototype.isActionAllowed = function (requestObject, context, kuzzle) {
   var
     deferred = q.defer();
-
   if (this.roles === undefined || this.roles.length === 0) {
     return q(false);
   }
 
-  async.some(this.roles, (role, callback) => {
-    role.isActionAllowed(requestObject, context, kuzzle)
-      .then(isAllowed => {
-        callback(isAllowed);
+  this.getRoles(kuzzle)
+    .then(roles => {
+      async.some(roles, (role, callback) => {
+        role.isActionAllowed(requestObject, context, kuzzle)
+          .then(isAllowed => {
+            callback(isAllowed);
+          });
+      }, result => {
```

Comment:
+1 @j33f 

[like said in the documentation](https://github.com/caolan/async/blob/v1.5.2/README.md#somearr-iterator-callback): 
**Note: the callbacks do not take an error as their first argument.**